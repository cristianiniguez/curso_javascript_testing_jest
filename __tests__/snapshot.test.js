import { getCharacter } from '../snapshot';
import rick from '../rick.json';

describe('Es hora de las instantaneas', () => {
  test('Snapshot', () => {
    expect(getCharacter(rick)).toMatchSnapshot();
  });
  test('Siempre fallara la instantanea', () => {
    const user = {
      createdAt: new Date(),
      id: Math.floor(Math.random() * 20),
    };
    expect(user).toMatchSnapshot();
  });
  test('Tenemos una excepción', () => {
    const user = {
      id: Math.floor(Math.random() * 20),
      name: 'Cristian Iñiguez',
    };
    expect(user).toMatchSnapshot({
      id: expect.any(Number),
    });
  });
});
