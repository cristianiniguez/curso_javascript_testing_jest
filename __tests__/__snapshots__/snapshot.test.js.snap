// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instantaneas Siempre fallara la instantanea 1`] = `
Object {
  "createdAt": 2020-10-30T15:59:24.981Z,
  "id": 13,
}
`;

exports[`Es hora de las instantaneas Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanches",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Es hora de las instantaneas Tenemos una excepción 1`] = `
Object {
  "id": Any<Number>,
  "name": "Cristian Iñiguez",
}
`;
